static/app/components/button.tsx

```diff
@@ -297,7 +309,7 @@ function BaseButton({
       <ButtonLabel size={size} borderless={borderless}>
         {icon && (
           <Icon size={size} hasChildren={hasChildren}>
-            {icon}
+            <IconDefaultsProvider size={ICON_SIZES[size]}>{icon}</IconDefaultsProvider>
```

Comment:
could update the component lib to demo this behavior 😉

Reply:
Will add it!